source/common/router/router.h

```diff
@@ -601,6 +613,15 @@ class Filter : Logger::Loggable<Logger::Id::router>,
   bool downstream_response_started_ : 1;
   bool downstream_end_stream_ : 1;
   bool is_retry_ : 1;
+  // mapped to the runtime guard "envoy.restart_features.ensure_connection_retry"
+  // on router creation.
+  bool ensure_connection_retry_ : 1;
+  // Set to true when the router is waiting to get an upcoming connection from the upstream,
```

Comment:
is this actually set true when the router is waiting on a connection?  It looks like it only does under certain circumstances (retry configured)

Reply:
That make sense, the comment was not clear at all, I updated it for 
```c++
  // Set to true when ensure_connection_retry_ above is true, retry on connection-failure is
  // enabled, and the router is waiting to get an upcoming connection from an upstream.
  // This will potentially increase the limit to which we are willing to buffer in order to ensure
  // we retry the connection failure.
```